docs/topics/db/sql.txt

```diff
@@ -4,8 +4,33 @@ Performing raw SQL queries
 
 .. currentmodule:: django.db.models
 
-When the :doc:`model query APIs </topics/db/queries>` don't go far enough, you
-can fall back to writing raw SQL. Django gives you two ways of performing raw
+.. Important:: 
+    
+    As of Django 2.1, there should be no reason to have to drop into raw SQL to 
+    perform complex queries. 
+
+    * Calculated fields can be created using :doc:`aggregations </topics/db/aggregation>` and `annotations`_.
+    * Obscure commands like explain plans, for example, have `native functions`_.
+    * Any other SQL command can be defined using :doc:`custom query expressions </ref/models/expressions/>`.
+
+.. _annotations: /topics/db/aggregation/#aggregating-annotations
+.. _native functions: /ref/models/querysets/#explain
+
+While there is no reason to use ``raw()`` any more, this documentation serves
```

Comment:
There is still some reasons and there will always be, there's only so much the ORM expressiveness can and will do.

Reply:
Hi @charettes, as per [trac](https://code.djangoproject.com/ticket/29940) given my research and confirming with multiple Django core devs leading up to and during DjangoCon US 2018, as of Django 2.1 there should be no reason to use `raw()` any more, period. I wanted to update the documentation (now 10 years out of date, with the warning about SQL injection only being added 5 years ago) to ensure that new developers are told of the recent updates since Django 2 which means there's no reason to drop into raw any more, but without removing the entire article about how it works.

Reply:
Hello @glasnt! I'll let other core developers chime in in this case but I feel like there might have been confusions between `raw()` and `extra()`. If we were to completely remove `raw()` there'd be no _blessed_ way of performing queries the ORM simply cannot express.

For example, the ORM still lacks support for common table expressions (AKA `WITH` queries), `LATERAL` joins, and other SQL features. It is condemned to lack behind in terms of features per it's abstracting nature over all the supported backends and the rate at which the specification evolves.

There's also the case of the multiple existing ORM bugs that can be worked around using `raw()`.

I really think that adjusting the documentation to mention that it's probably not necessary to use `raw()` is a step forward. I just don't think we'll ever be able to express any form of queries using the ORM.

Reply:
Agreed. If there's no reason for use it anymore, then we'd deprecate it. I don't see that happening.